controllers/wavefrontalert_controller.go

```diff
@@ -160,9 +282,28 @@ func (r *WavefrontAlertReconciler) HandleDelete(ctx context.Context, wfAlert *al
 	return nil
 }
 
+//convertAlertCR converts alert CR to wf.Alert
+func (r *WavefrontAlertReconciler) convertAlertCR(ctx context.Context, wfAlert *alertmanagerv1alpha1.WavefrontAlert, alert *wf.Alert) {
+	log := log.Logger(ctx, "controllers", "wavefrontalert_controller", "convertAlertCR")
+	log = log.WithValues("wavefrontalert_cr", wfAlert.Name, "namespace", wfAlert.Namespace)
+	if err := wavefront.ConvertAlertCRToWavefrontRequest(ctx, wfAlert.Spec, alert); err != nil {
+		errMsg := "unable to convert the wavefront spec to Alert API request. will not be retried"
+		log.Error(err, errMsg)
+		r.Recorder.Event(wfAlert, v1.EventTypeWarning, "MalformedSpec", errMsg)
+		wfAlert.Status = alertmanagerv1alpha1.WavefrontAlertStatus{
+			RetryCount:       wfAlert.Status.RetryCount + 1,
+			ErrorDescription: errMsg,
+			State:            alertmanagerv1alpha1.MalformedSpec,
+		}
+		// There is no use of requeue in this case
+		r.CommonClient.UpdateStatus(ctx, wfAlert, alertmanagerv1alpha1.MalformedSpec)
```

Comment:
Just to confirm, there will be no requeue when you update status unless you pass the flag right? Because of this [code](https://github.com/keikoproj/alert-manager/blob/fa580cb4e7a6651f26a311026daf10e9780085b3/controllers/common/common.go#L42)

Reply:
Yes